examples/shader/render_event.rs

```diff
@@ -0,0 +1,90 @@
+//! Simple example demonstrating the use of [`App::init_render_event`] to send events from the
+//! render world to the main world
+
+use bevy::{
+    diagnostic::FrameCount,
+    ecs::system::{LocalBuilder, ParamBuilder},
+    prelude::*,
+    render::{
+        render_event::{MainEventWriter, RenderEventApp},
+        Extract, Render, RenderApp,
+    },
+};
+
+fn main() -> AppExit {
+    App::new()
+        .add_plugins((DefaultPlugins, RenderEventDemoPlugin))
+        .run()
+}
+
+// We need a plugin to organize all the systems and render node required for this example
+struct RenderEventDemoPlugin;
+impl Plugin for RenderEventDemoPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_render_event::<FrameRenderedEvent>()
+            .add_systems(Update, read_render_event);
+    }
+
+    fn finish(&self, app: &mut App) {
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        // Since `FrameCount` is not present in the render world by default,
+        // we need to initialize it here. This is specific to this example,
+        // and should be unnecessary for most users.
+        render_app.init_resource::<FrameCount>();
+
+        let send_render_event = (
+            ParamBuilder,
+            ParamBuilder,
+            ParamBuilder,
+            LocalBuilder(Timer::from_seconds(2.0, TimerMode::Repeating)),
+        )
+            .build_state(render_app.world_mut())
+            .build_system(send_render_event);
```

Comment:
This feels a bit unnecessarily complex. I'd suggest using a resource instead just because that's a more common ECS api.